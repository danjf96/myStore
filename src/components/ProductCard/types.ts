import { PRODUCTSTATE } from "../../store/ducks/products/types";

export interface ProductsCardProps extends PRODUCTSTATE {
    number?: number,
    quantity?: boolean,
    onPressQuantity?: (type: 'increase' | 'decrease') => void,
    testId?: string
}