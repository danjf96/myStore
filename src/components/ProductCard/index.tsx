import React from 'react'
import { Image, Text, TouchableOpacity, View } from 'react-native'
import { formatMoney } from '../../utils/mask'
import Styles from './styles'
import { ProductsCardProps } from './types'

const ProductCard = ({
    name,
    image,
    price,
    stock,
    quantity = false,
    number = 0,
    onPressQuantity = () => null,
    testId,
    styleContainer,
    onPress
}:ProductsCardProps) => {
    const uri = image?.replace('http://lorempixel.com/','https://loremflickr.com/')

    return (
        <TouchableOpacity disabled={quantity} onPress={onPress} style={{...Styles.containerCard, ...styleContainer}} testID={testId}>
            <TouchableOpacity disabled={!quantity} onPress={onPress} style={Styles.containerButton}>
                <Image source={{ uri }} style={Styles.image} />

                <View>
                    <Text style={Styles.name}>{name}</Text>
                    <Text style={Styles.price}>R$ {formatMoney(price)}</Text>
                    {/* <Text style={Styles.name}>Disponivel: {stock}</Text> */}
                </View>
            </TouchableOpacity>

            {quantity && <View style={{ flexDirection: 'row', alignItems: 'center', marginLeft: 15 }}>

                <TouchableOpacity onPress={ () => onPressQuantity('decrease')} style={{ padding: 10 }}>
                    <Text style={{ fontSize: 30 }}>-</Text>
                </TouchableOpacity>

                <Text style={{  padding: 8, borderWidth:1, borderRadius: 5 }}>{number}</Text>

                <TouchableOpacity onPress={ () => onPressQuantity('increase')}  style={{ padding: 10 }}>
                    <Text style={{ fontSize: 30 }}>+</Text>
                </TouchableOpacity>

            </View>}

        </TouchableOpacity>
    )
}

export default ProductCard