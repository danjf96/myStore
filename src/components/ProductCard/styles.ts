import { Platform, StyleSheet } from "react-native";
import Colors from "../../assets/Colors";

const Styles = StyleSheet.create({
    containerCard: {
        width: '100%',
        padding: 10,
        flexDirection: 'row',
        alignItems: 'center',
        //justifyContent: 'space-between'
        borderRadius: 10,
        elevation: 2,
        backgroundColor: 'white',
        ...Platform.OS == 'ios' ?
        {
            shadowColor: Colors.backgroundModal,
            shadowOffset: { 
                width: 0, 
                height: 4 
            },
            shadowOpacity: 0.2,
            shadowRadius: 2
        } : {}
    },

    containerButton: {
        flexDirection: 'row',
        alignItems: 'center',
    },

    image: {
        width: 60,
        height: 60,
        borderRadius: 10,
        marginRight: 15
    },

    name: {
        fontSize: 18,
        marginBottom: 5,
        color: 'black'
    },

    price: {
        fontSize: 18,
        fontWeight: 'bold',
        color: 'black'
    }


})

export default Styles