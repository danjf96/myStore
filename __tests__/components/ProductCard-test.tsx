/**
 * @format
 */

 import 'react-native';
 import React from 'react';
 import ProductCard from '../../src/components/ProductCard';
import moment from 'moment';
import { fireEvent, render } from 'react-native-testing-library';
const onPress = jest.fn();

describe('Render component', () => {
    const fakeValue = { name:'test', image:'http://lorempixel.com/640/480/food', price: 100, stock: 1, createdAt: moment().format('DD/MM/YYYY'), id:'1'}
    it('renders correctly', async () => { 
        const screen = render( <ProductCard {...fakeValue}/>)
        expect(screen.toJSON()).toMatchSnapshot()
    });

    test('click card', () => {
        const onPress = jest.fn();
        const { getByTestId } = render( <ProductCard testId='product' onPress={onPress} {...fakeValue}/>)
        fireEvent(getByTestId('product'), 'press')
        expect(onPress).toBeCalled()
    })

    test('change quantity', () => {
        const onPress = jest.fn();
        const onPressQuantity = jest.fn();
        const { getByText } = render( <ProductCard 
            number={0} testId='product' 
            onPress={onPress} {...fakeValue}
            onPressQuantity={onPressQuantity}
            quantity={true}
        />)

        fireEvent(getByText('+'), 'press')
        expect(onPressQuantity).toBeCalled()

        fireEvent(getByText('-'), 'press')
        expect(onPressQuantity).toBeCalled()
    })

})



   
 
   