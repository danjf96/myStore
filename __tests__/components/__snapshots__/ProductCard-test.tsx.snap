// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render component renders correctly 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": false,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 10,
      "elevation": 2,
      "flexDirection": "row",
      "justifyContent": "flex-start",
      "opacity": 1,
      "padding": 10,
      "shadowColor": "#353535",
      "shadowOffset": Object {
        "height": 4,
        "width": 0,
      },
      "shadowOpacity": 0.2,
      "shadowRadius": 2,
      "width": "100%",
    }
  }
>
  <View
    accessibilityState={
      Object {
        "disabled": true,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "opacity": 1,
        "width": "100%",
      }
    }
  >
    <Image
      source={
        Object {
          "uri": "https://loremflickr.com/160/120/food",
        }
      }
      style={
        Object {
          "borderRadius": 10,
          "height": 60,
          "marginRight": 15,
          "width": 60,
        }
      }
    />
    <View>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "black",
            "fontSize": 18,
            "marginBottom": 5,
          }
        }
      >
        test
      </Text>
      <Text
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "black",
            "fontSize": 18,
            "fontWeight": "bold",
          }
        }
      >
        R$ 
        100,00
      </Text>
    </View>
  </View>
</View>
`;
